R/eval-walk.R

```diff
@@ -113,6 +121,12 @@ ensure_named <- function(pos,
   pos
 }
 
+check_empty <- function(x, allow_empty = TRUE, call = caller_env()) {
```

Comment:
Nit: Though it reduces some repetition here, it seems unusual for a `check_` function to take a boolean variable disabling the check. I think the general pattern is more like

```r
if (check) {
  check_foo(x)
}
```